test/test-files/replace_by_env_in_file.conf

```diff
@@ -0,0 +1,6 @@
+key_1 = value
```

Comment:
I would probably move this into `test/config/` sub-folder instead. Look at the config overrides test that does similar. It's better for scoping a file that belongs only to the test I think? Especially when they're meant to be config files.

Reply:
Technically if you retain the separate key/value lines per test like the original, then you could set all ENV and run a single test in the container. Each `--env` arg can be pre-faced with a comment for the test-case description.

Then again individual test cases are nice too :+1:

Reply:
Something similar done in 1849c7182a8b4b0697e47dc46b7ab5bc690fefcf

Reply:
Not exactly what I meant :sweat_smile: 

I was thinking about a single container with `--env` for each test case, the input config, and then comparing the transformed config in the container to some "expected" version of the config.

Without the container like that, the separate test cases seemed fine, and were a bit more descriptive about each test, now they're more vague and I'd need to reference the config a bit more to make sense of what the lines are actually testing that differentiates them from the others in the same test case group.